### PR TITLE
Beta docs restructure image service procedure

### PR DESCRIPTION
**Pull request by klgill**
I divided the "Adopting the Image service" procedure into an assembly with one procedure per backend, and a verification procedure.

@konan-abhi @fmount Can you please review the structure of the glance chapter? I have a few comments that begin with "kgilliga" in the file as well.

Writers: Please use the peer review checklist to stay in scope: https://docs.google.com/document/d/1HijeVeppLEQot6AOgxhzy-UBeu-_WKwrnWRSxQoQ9jY/edit#heading=h.o8ezt27iw7kl

### CODE REVIEW

File: docs_user/modules/proc_verifying-the-image-service-adoption.adoc [hunk 1]
```diff
@@ -0,0 +1,128 @@
+[id="verifying-the-image-service-adoption_{context}"]
+
+= Verifying the {image_service} adoption
+
+Verify that you successfully adopted your {image_service_first_ref} to the {rhos_long} {rhos_curr_ver} deployment.
+//kgilliga: Does this procedure apply to all backends, or do some verification steps only apply to specific backends?
```
Comment (fmount) [inline docs_user/modules/proc_verifying-the-image-service-adoption.adoc:6]: The example points to a Ceph backend, but in  general I think the steps are the same for all the backends (except when you check the existing data on the Ceph cluster).
We can change the title and point to the fact this is an example with a Ceph backend.
Is that the goal?

File: docs_user/modules/proc_verifying-the-image-service-adoption.adoc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+. Test the glance service from the {rhos_prev_long} CLI. You can compare and make sure the configuration has been correctly applied to the glance pods:
++
+----
+./os-diff cdiff --service glance -c /etc/glance/glance.conf.d/02-config.conf  -o glance_patch.yaml --frompod -p glance-api
+----
++
+If no line appears, then the configuration is correctly done.
+
+. Inspect the resulting glance pods and ensure that Ceph secrets are properly mounted:
```
Comment (fmount) [inline docs_user/modules/proc_verifying-the-image-service-adoption.adoc:18]: I think this line (L18) should be moved to L34, where we actually check the Ceph bits, and yeah, checking the Ceph secret is an action you won't do for other backends.

File: docs_user/modules/proc_verifying-the-image-service-adoption.adoc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+//kgilliga: Does this step only apply to customers who are using a Ceph backend?
```
Comment (fmount) [inline docs_user/modules/proc_verifying-the-image-service-adoption.adoc:19]: The step from L20 to L33 is valid for every backend, as it's checking the resulting glance config in the running Pod.

File: docs_user/modules/proc_verifying-the-image-service-adoption.adoc [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+----
+(openstack)$ image list
++--------------------------------------+--------+--------+
+| ID                                   | Name   | Status |
++--------------------------------------+--------+--------+
+| c3158cad-d50b-452f-bec1-f250562f5c1f | cirros | active |
++--------------------------------------+--------+--------+
+----
+
+. Test that an image can be created on the adopted service:
```
Comment (fmount) [inline docs_user/modules/proc_verifying-the-image-service-adoption.adoc:65]: This step tests the Image service is working by uploading an image. Not sure we need it downstream.

File: docs_user/modules/proc_verifying-the-image-service-adoption.adoc (hunk 1 above)
Comment (klgill) [reply docs_user/modules/proc_verifying-the-image-service-adoption.adoc:6]: @fmount The goal was to have a general verification procedure that applies to all backends. However, another option is to add a ".Verification" heading in each specific backend procedure and only include the steps that apply to that backend. If some of these steps apply to all backends, the steps will be repeated in each procedure, but that's okay.

Comment (klgill) [reply docs_user/modules/proc_verifying-the-image-service-adoption.adoc:6]: We can keep the Ceph bits, and I can rewrite the step to say something like: "If you use a Ceph backend, <do this>."

File: docs_user/modules/proc_adopting-image-service-with-block-storage-backend.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+          cinder_store_password = {{ .ServicePassword }}
+          cinder_store_project_name = service
+          cinder_catalog_info = volumev3::internalURL
+          cinder_use_multipath = true
+----
+
+.Prerequisites
+
+* Previous Adoption steps completed. Notably, MariaDB, Keystone and Barbican
+should be already adopted.
```
Comment (konan-abhi) [inline docs_user/modules/proc_adopting-image-service-with-block-storage-backend.adoc:31]: Shouldn't Cinder also be adopted before this?

File: docs_user/modules/proc_adopting-image-service-with-block-storage-backend.adoc [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+Having {block_storage} as a backend establishes a dependency between the two services, and any deployed `GlanceAPI` instance would not work if the {image_service} is configured with {block_storage} that is still not available in the `OpenStackControlPlane`.
+Once {block_storage}, and in particular `CinderVolume`, has been adopted through the procedure described in <<adopting-the-block-storage-service>>, it is possible to proceed with the `GlanceAPI` adoption.
+
+. Verify that `CinderVolume` is available:
++
+----
+$ oc get pod -l component=cinder-volume | grep Running
+cinder-volume-75cb47f65-92rxq   3/3     Running   0
+----
+
```
Comment (konan-abhi) [inline docs_user/modules/proc_adopting-image-service-with-block-storage-backend.adoc:88]: I think line 82~88 should be moved after line 31

File: docs_user/modules/proc_adopting-image-service-with-object-storage-backend.adoc [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+          swift_store_endpoint_type = internalURL
+          swift_store_user = service:glance
+          swift_store_key = {{ .ServicePassword }}
+----
+
+.Prerequisites
+
+* Previous Adoption steps completed. Notably, MariaDB, Keystone and Barbican
+should be already adopted.
+
```
Comment (konan-abhi) [inline docs_user/modules/proc_adopting-image-service-with-object-storage-backend.adoc:30]: This should also include swift IMO

File: docs_user/modules/proc_adopting-image-service-with-object-storage-backend.adoc [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+Having {object_storage} as a backend establishes a dependency between the two services, and any deployed `GlanceAPI` instance would not work if {image_service} is configured with {object_storage} that is still not available in the `OpenStackControlPlane`.
+Once {object_storage}, and in particular `SwiftProxy`, has been adopted through the procedure described in <<adopting-object-storage-service>>, it is possible to proceed with the `GlanceAPI` adoption.
+
+. Verify that `SwiftProxy` is available:
++
+----
+$ oc get pod -l component=swift-proxy | grep Running
+swift-proxy-75cb47f65-92rxq   3/3     Running   0
+----
+
```
Comment (konan-abhi) [inline docs_user/modules/proc_adopting-image-service-with-object-storage-backend.adoc:84]: Line 77 ~ 84 should be moved after line 30

**Comment by klgill** [comment]
> LGTM. I think if we want factual changes to the instructions rather than stylistic restructuring, they could also be made by the storage team in a follow-up PR?

@jistr I'm okay with merging this and addressing Abhishek's questions in a separate PR. I'm not sure who can best respond. @fmount ?

**Comment by konan-abhi** [comment]
> > LGTM. I think if we want factual changes to the instructions rather than stylistic restructuring, they could also be made by the storage team in a follow-up PR?
> 
> @jistr I'm okay with merging this and addressing Abhishek's questions in a separate PR. I'm not sure who can best respond. @fmount ?

I am ok to address this in a followup PR.

**Review by jistr — APPROVED** [review]

**Comment by jistr** [comment]
Thank you folks, merging.

**Comment by fmount** [comment]
@klgill sorry  /me terribly late, +1 to follow up on what @konan-abhi pointed out!

**Pull request merged**